tests/base/src/test/java/org/wildfly/security/http/impl/AbstractBaseHttpTest.java

```diff
@@ -459,23 +464,40 @@ protected CallbackHandler getCallbackHandler(String username, String realm, Stri
                         throw new IllegalStateException(e);
                     }
                 } else if (callback instanceof EvidenceVerifyCallback) {
-                    PasswordGuessEvidence evidence = (PasswordGuessEvidence) ((EvidenceVerifyCallback) callback).getEvidence();
-                    ((EvidenceVerifyCallback) callback).setVerified(Arrays.equals(evidence.getGuess(), password.toCharArray()));
-                    evidence.destroy();
+                    if(((EvidenceVerifyCallback) callback).getEvidence() instanceof PasswordGuessEvidence){
+                        PasswordGuessEvidence evidence = (PasswordGuessEvidence) ((EvidenceVerifyCallback) callback).getEvidence();
+                        ((EvidenceVerifyCallback) callback).setVerified(Arrays.equals(evidence.getGuess(), password.toCharArray()));
+                        evidence.destroy();
+                    }else if(((EvidenceVerifyCallback) callback).getEvidence() instanceof BearerTokenEvidence){
+                        BearerTokenEvidence evidence = (BearerTokenEvidence) ((EvidenceVerifyCallback) callback).getEvidence();
+                        ((EvidenceVerifyCallback) callback).setVerified(Objects.equals(token, evidence.getToken()));
+                    }
                 } else if (callback instanceof AuthenticationCompleteCallback) {
                     // NO-OP
                 } else if (callback instanceof IdentityCredentialCallback) {
                     Credential credential = ((IdentityCredentialCallback) callback).getCredential();
-                    MatcherAssert.assertThat(credential, CoreMatchers.instanceOf(PasswordCredential.class));
-                    ClearPassword clearPwdCredential = ((PasswordCredential) credential).getPassword().castAs(ClearPassword.class);
-                    Assert.assertNotNull(clearPwdCredential);
-                    Assert.assertArrayEquals(password.toCharArray(), clearPwdCredential.getPassword());
+                    if(token!=null) {
```

Comment:
@keshav-redhat Just a small nitpick, try to use similar style in your code to the rest of the code. Eg. use spaces around `!=` and after `, `, `}` etc.